orc8r/cloud/go/services/ctraced/obsidian/handlers/gateway_api.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"magma/orc8r/cloud/go/orc8r"
+	"magma/orc8r/cloud/go/services/configurator"
+	"magma/orc8r/cloud/go/services/dispatcher/gateway_registry"
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+)
+
+type GWCtracedClient interface {
+	StartCallTrace(networkId string, gatewayId string, req *protos.StartTraceRequest) (*protos.StartTraceResponse, error)
+
+	EndCallTrace(networkId string, gatewayId string, req *protos.EndTraceRequest) (*protos.EndTraceResponse, error)
+}
+
+type GWCtracedClientImpl struct{}
+
+func NewGwCtracedClientImpl() GWCtracedClientImpl {
```

Comment:
The type can be private and we can return the interface here.

Reply:
Okay